File: encapsulamiento/sales-csharp/Course.OOP.Encapsulation.Business.Other/AnotherInvoice.cs
using System;

namespace Course.OOP.Encapsulation.Business.Other
{
    class AnotherInvoice: Invoice
    {
        void MyMethod() {
            decimal result = this.CalculateTotal();
            //Si descomentas la siguiente verás que el método no se puede acceder.
            //this.OtherMethod();
        }
    }
}

File: encapsulamiento/sales-java/src/main/java/course/oop/encapsulation/other/AnotherInvoice.java
package course.oop.encapsulation.other;

import course.oop.encapsulation.Invoice;

import java.math.BigDecimal;

public class AnotherInvoice extends Invoice {

    void myMethod() {
        BigDecimal amount = calculateTotal();
        this.addItem("ab", new BigDecimal(1));
        System.out.println(amount);
        //this.items;
    }

}
